src/Rings/MPolyQuo.jl

```diff
@@ -807,9 +842,20 @@ true
 """
 function ==(f::MPolyQuoRingElem{T}, g::MPolyQuoRingElem{T}) where T
   check_parent(f, g)
-  simplify(f)
-  simplify(g)
-  return f.f == g.f
+  Q = parent(f)::MPolyQuoRing
+  P = base_ring(Q)::MPolyRing
+  kk = coefficient_ring(P)::Ring
+
+  # The following bracket is for the default assumption
+  if is_coefficient_type_with_groebner_basis_algorithm_in_singular(kk)
+    f.f == g.f && return true
+    hash(f) == hash(g) && return true # calls simplify already
+    return f.f == g.f
```

Comment:
This looks wrong to me. Please double check

Reply:
Indeed. What was I thinking...? Corrected in #2789 .